retriever_test.go

```diff
@@ -0,0 +1,237 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"github.com/h2non/gock"
+	"github.com/nbio/st"
+	"net/http"
+	"os"
+	"testing"
+)
+
+func TestFileBasedRetrieve_Success(t *testing.T) {
+	inputFilePath := "retriever_test"
+	inputFile, err := os.OpenFile(inputFilePath, os.O_RDWR|os.O_CREATE, 0600)
+	if err != nil {
+		t.Errorf("Failed opening test file=%s, %s", inputFilePath, err)
+		return
+	}
+	defer os.Remove(inputFilePath)
+
+	var expectedIds []string
+	expectedIds = append(expectedIds, "2d3e16e0-61cb-4322-8aff-3b01c59f4daa", "fd4459b2-cc4e-4ec8-9853-c5238eb860fb")
+
+	for _, id := range expectedIds {
+		inputFile.WriteString(id)
+		inputFile.WriteString("\n")
+	}
+	inputFile.Close()
+
+	retriever := newFileBasedIDListRetriever(inputFilePath)
+	var idsChan = make(chan string)
+	var errChan = make(chan error)
+	var actualIds = make(map[string]struct{})
+
+	go retriever.Retrieve(idsChan, errChan)
+
+	for idsChan != nil {
+		select {
+		case id, ok := <-idsChan:
+			if !ok {
+				idsChan = nil
+			} else {
+				actualIds[id] = struct{}{}
+			}
+		case err := <-errChan:
+			t.Errorf("Error not expected: %s", err)
+			return
+		}
+	}
+
+	for _, id := range expectedIds {
+		if _, found := actualIds[id]; !found {
+			st.Expect(t, true, found)
+			return
+		}
+	}
+}
+
+func TestFileBasedRetrieve_FileOpenFailure(t *testing.T) {
+	expectedError := errors.New("ERROR - Failed opening file=non_existing_file: open non_existing_file: The system cannot find the file specified.")
```

Comment:
It could come same other error messages from different OSes - could you please check only the part we set up for the error?